scale/job/views.py

```diff
@@ -200,6 +221,23 @@ def patch(self, request, job_type_id):
         except InvalidInterfaceDefinition as ex:
             raise BadParameter('Job type interface invalid: %s' % unicode(ex))
 
+        # Validate the job configuration and pull out secrets
+        configuration_dict = rest_util.parse_dict(request, 'configuration', required=False)
+        configuration = None
+        secrets = None
+        try:
+            if configuration_dict:
+                configuration = JobConfiguration(configuration_dict)
+                if interface:
+                    secrets = configuration.get_secret_settings(interface)
+                    configuration.validate(interface)
+                else:
+                    stored_interface = JobType.objects.values_list('interface', flat=True).get(pk=job_type_id)
+                    secrets = configuration.get_secret_settings(stored_interface)
+                    configuration.validate(JobInterface(stored_interface))
```

Comment:
These two lines can be pulled out of the if statement:
```
secrets = configuration.get_secret_settings(interface)
configuration.validate(interface)
```

Your if statement could simply check for `interface` and if None do the query to populate it.